lib/internal/test_runner/test.js

```diff
@@ -60,6 +61,7 @@ const testOnlyFlag = !isTestRunner && getOptionValue('--test-only');
 const kShouldAbort = Symbol('kShouldAbort');
 const kRunHook = Symbol('kRunHook');
 const kHookNames = ObjectSeal(['before', 'after', 'beforeEach', 'afterEach']);
+const kUnwrapErrors = new SafeSet([kTestCodeFailure, kHookFailure, 'uncaughtException', 'unhandledRejection']);
```

Comment:
https://github.com/nodejs/node/blob/81a68b37cfc9be0170e31349281b15b14f19959f/doc/contributing/primordials.md#L399-L400

E.g. it can crash the process with the following setup:

```sh
echo 'delete Array.prototype[Symbol.iterator]' > delete.cjs
out/Release/node -r ./delete.cjs test/fixtures/test-runner/test/random.cjs
```

We can protect from this by calling `.add` several times:

```suggestion
const kUnwrapErrors = new SafeSet()
  .add(kTestCodeFailure).add(kHookFailure)
  .add('uncaughtException').add('unhandledRejection');
```

or by using `SafeArrayIterator`, but that's known to have bad perf implications.

Reply:
I would expect `--require` or `--import` to only run in the inner proccess, but I will fix anyway